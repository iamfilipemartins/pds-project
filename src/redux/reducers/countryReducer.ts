import {
  ICountrySelected,
  SET_COUNTRY_SELECTED,
  SET_COUNTRY_DETAILS,
  ICountryDetails,
} from '../actions/countryActions';

type CountryNameState = {
  countrySelected: ICountrySelected;
  countryDetails: ICountryDetails;
};

const initialState: CountryNameState = {
  countrySelected: {
    mapData: {
      ABBREV: '',
      CONTINENT: '',
      FORMAL_EN: '',
      GDP_MD_EST: 0,
      GDP_YEAR: 0,
      ISO_A2: '',
      ISO_A3: '',
      NAME: '',
      NAME_LONG: '',
      POP_EST: 0,
      POP_RANK: 0,
      POP_YEAR: 0,
      REGION_UN: '',
      SUBREGION: '',
    },
  },
  countryDetails: {
    id: { M49: 0, ISO_3166_1_ALPHA_2: '', ISO_3166_1_ALPHA_3: '' },
    nome: { abreviado: '' },
    area: { total: '', unidade: { nome: '', sÃ­mbolo: '', multiplicador: 1 } },
    localizacao: {
      regiao: { id: { M49: 0 }, nome: '' },
      sub_regiao: { id: { M49: 0 }, nome: '' },
      regiao_intermediaria: { id: { M49: 0 }, nome: '' },
    },
    linguas: [],
    governo: { capital: { nome: '' } },
    unidades_monetarias: [],
    historico: '',
  },
};

const countryReducer = (state: CountryNameState = initialState, action: any): any => {
  switch (action.type) {
    case SET_COUNTRY_SELECTED:
      return {
        ...state,
        countrySelected: action.payload,
      };
    case SET_COUNTRY_DETAILS:
      return {
        ...state,
        countryDetails: action.payload,
      };
    default:
      return state;
  }
};
export default countryReducer;
