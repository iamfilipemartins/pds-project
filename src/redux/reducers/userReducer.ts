import { IUserData, SET_EMAIL_RECOVERY_DATA, SET_LOGIN_DATA, SET_SIGNUP_DATA } from '../actions/userActions';

type UserState = {
  user: IUserData;
};

const initialState: UserState = {
  user: {
    login: {},
    signup: {},
    emailRecovery: ''
  }
};
const userReducer = (state: UserState = initialState, action: any) : any => {
  switch (action.type) {
    case SET_LOGIN_DATA:
      return {
        ...state,
        user: {
          ...state.user,
          login: action.payload,
        },
      };
      case SET_EMAIL_RECOVERY_DATA:
        return {
          ...state,
          user: {
            ...state.user,
            emailRecovery: action.payload,
          },
        };
        case SET_SIGNUP_DATA:
          return {
            ...state,
            user: {
              ...state.user,
              signup: action.payload,
            },
          };
    default:
      return state;
  }
};

export default userReducer;
