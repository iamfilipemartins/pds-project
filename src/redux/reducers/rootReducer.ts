import { combineReducers } from 'redux';
import countryReducer from './countryReducer';

const rootReducer = combineReducers({
  country: countryReducer,
});
export type AppState = ReturnType<typeof rootReducer>;
export default rootReducer;
