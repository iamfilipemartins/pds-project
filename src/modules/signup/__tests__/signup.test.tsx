/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Signup from '..';
import { COUNTRY_DETAILS_INITIAL_STATE, COUNTRY_SELECTED_INITIAL_STATE } from '../../../redux/reducers/countryReducer';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router', () => ({
  ...(jest.requireActual('react-router') as any),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return () => {};
    }),
    useDispatch: jest.fn().mockImplementation(() => {
      return () => {};
    }),
  };
});

describe('<Signup />', () => {
  window.scrollTo = jest.fn();

  const initialState = {
    country: {
      countrySelected: COUNTRY_SELECTED_INITIAL_STATE,
      countryDetails: COUNTRY_DETAILS_INITIAL_STATE,
    },
    user: {
      login: {},
      signup: {},
      emailRecovery: '',
    },
  };
  const mockStore = configureStore();
  test('should display a blank signup form, with remember me checked by default', async () => {
    const store = mockStore(initialState);
    const { findByTestId } = render(
      <Provider store={store}>
        <Signup />
      </Provider>
    );

    const loginScreen = await findByTestId('signup');

    expect(loginScreen).toMatchSnapshot();
  });

  test('should allow entering a email and an password', async () => {
    const store = mockStore(initialState);
    const { findByTestId } = render(
      <Provider store={store}>
        <Signup />
      </Provider>
    );
    const email = (await findByTestId('email')) as HTMLInputElement;
    const password = (await findByTestId('password')) as HTMLInputElement;
    const loginScreen = (await findByTestId('signup')) as HTMLInputElement;

    fireEvent.change(email, { target: { value: 'test@test.com' } });
    fireEvent.change(password, { target: { value: 'password' } });

    expect(loginScreen).toBeDefined();
    expect(email).toBeDefined();
    expect(email.value).toEqual('test@test.com');
    expect(password).toBeDefined();
    expect(password.value).toEqual('password');
  });
});
