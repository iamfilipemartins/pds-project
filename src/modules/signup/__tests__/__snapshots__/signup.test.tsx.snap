// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> should display a blank signup form, with remember me checked by default 1`] = `
<div
  class="sc-gsDKAQ iFzJiF"
  data-testid="signup"
>
  <div
    class="sc-dkPtRN crvUag"
    width="1024"
  >
    <img
      class="sc-bdvvtL"
    />
    <h1
      class="sc-hKwDye gGnQLo"
      width="1024"
    >
      Geolog
    </h1>
  </div>
  <div
    class="sc-furwcr anivc"
    width="1024"
  >
    <h1
      class="sc-jrQzAO gfpuAK"
    >
      Criar conta
    </h1>
    <div
      class="sc-kDTinF bInWyk"
    >
      <p
        class="sc-crHmcD dKnwJW"
      >
        Email *
      </p>
      <input
        class="sc-iqseJM dhtnqU"
        data-testid="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="sc-kDTinF bInWyk"
    >
      <p
        class="sc-crHmcD dKnwJW"
      >
        Senha *
      </p>
      <input
        class="sc-iqseJM dhtnqU"
        data-testid="password"
        type="password"
        value=""
      />
    </div>
    <div
      class="sc-iCfMLu iBhzdD"
      width="1024"
    >
      <div
        class="sc-egiyK"
      >
        <button
          class="sc-bqiRlB dpOhwe"
          color="#FFFFFF"
          disabled=""
        >
          Cadastrar
        </button>
      </div>
    </div>
  </div>
  <div
    class="sc-eCImPb iCHfyu"
    width="1024"
  >
    <p
      class="sc-jRQBWg dEaWjf"
    >
      JÃ¡ possui uma conta?
    </p>
    <p
      class="sc-gKclnd kKqYGw"
    >
      Entrar
    </p>
  </div>
</div>
`;
